.github/workflows/soaking.yml

```diff
@@ -1,160 +1,139 @@
-        if: ${{ success() }}
-        run: |
-          endpoint=$(cat endpoint_file)
-          cp sample-apps/python-lambda/tools/lambdaExpectedTrace.mustache test-framework/validator/src/main/resources/expected-data-template/
-          cd test-framework
-          ./gradlew :validator:run  --args="-c default-lambda-validation.yml --endpoint $endpoint --region $AWS_REGION"
-
-      - name: turn on Lambda Insight
-        working-directory: ${{ env.WORK_DIR }}
-        if: ${{ success() }}
-        run: |
-          ./updateFunction.sh -a "arn:aws:lambda:us-west-2:580247275435:layer:LambdaInsightsExtension:10"
-          sleep 30
-
-      - name: manually soaking test
-        working-directory: ${{ env.WORK_DIR }}
-        if: ${{ success() && github.event_name == 'workflow_dispatch' }}
-        run: |
-          pip install boto3
-          functionName=$(./distribute.sh -f)
-          python3 soaking.py $functionName ${{ github.event.inputs.args }}
-          echo "Soaking test done."
-      
-      - name: nightly soaking test
-        working-directory: ${{ env.WORK_DIR }}
-        if: ${{ success() && github.event_name != 'workflow_dispatch' }}
-        run: |
-          pip install boto3
-          functionName=$(./distribute.sh -f)
-          python3 soaking.py $functionName 18000 5
-          echo "Soaking test done." 
-      
-      - name: integration test
-        if: ${{ success() }}
-        run: |
-          endpoint=$(cat endpoint_file)
-          cp sample-apps/python-lambda/tools/lambdaExpectedTrace.mustache test-framework/validator/src/main/resources/expected-data-template/
-          cd test-framework
-          ./gradlew :validator:run  --args="-c default-lambda-validation.yml --endpoint $endpoint --region $AWS_REGION"
-
-      - name: clean
-        working-directory: ${{ env.WORK_DIR }}
-        if: ${{ success() }}
+          registry: public.ecr.aws
+      - name: Build layers / functions
+        run: ${{ matrix.build_command }}
+        working-directory: ${{ matrix.build_directory }}
+      - name: Build collector
+        run: make package
+        working-directory: opentelemetry-lambda/collector
+      - uses: hashicorp/setup-terraform@v1
+      - name: Initialize terraform
+        run: terraform init
+        working-directory: ${{ matrix.terraform_directory }}
+      - name: Apply terraform
+        run: terraform apply -auto-approve
+        working-directory: ${{ matrix.terraform_directory }}
+        env:
+          # TODO(anuraaga): Consider picking a canonical language to manage the pushed collector layer and use different
+          # names for others. Won't be needed if we don't publish separate collector layer.
+          TF_VAR_collector_layer_name: aws-otel-lambda-collector-${{ github.sha }}
+          TF_VAR_sdk_layer_name: aws-otel-lambda-${{ layer_type }}-${{ github.sha }}
+          TF_VAR_function_name: hello-lambda-${{ matrix.language }}-${{ github.run_id }}-${{ matrix.name }}
```

Comment:
Do you want to support both SAM and TF in soak and build-main workflows? or TF only?